### PR TITLE
fix(oauth): Adds DynamicParameter --userInfoRequirements to enable do…

### PR DESCRIPTION
**Pull request by ttomsu**
…main restrictions.

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -633,6 +633,7 @@ hal config security oauth2 edit [parameters]
  * `--no-validate`: (*Default*: `false`) Skip validation.
  * `--preEstablishedRedirectUri`: The externally accessible URL for Gate. For use with load balancers that do any kind of address manipulation for Gate traffic, such as an SSL terminating load balancer.
  * `--provider`: The OAuth provider handling authentication. The supported options are Google, GitHub, and Azure
+ * `--userInfoRequirements`: (*Default*: `OAuth2.UserInfoRequirements()`) This map of requirements the userInfo request must have. This is used to restrict user login to specific domains or having a specific attribute.
```
Comment (duftler) [inline docs/commands.md]: Should this read 'The map of...'?

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java [hunk 2]
```diff
@@ -56,6 +57,13 @@
   )
   private String preEstablishedRedirectUri;
 
+  @DynamicParameter(
```
Comment (lwander) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java]: How is this parsed? The above generated help text above doesn't make this too clear

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java [hunk 3]
```diff
@@ -74,6 +82,10 @@ protected AuthnMethod editAuthnMethod(OAuth2 authnMethod) {
 
     authnMethod.setProvider(provider != null ? provider : authnMethod.getProvider());
 
+    if (!userInfoRequirements.isEmpty()) {
```
Comment (lwander) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java]: By supplying the default `new OAuth2.UserInfoRequirements()` to this parameter, how will this be empty?

File: docs/commands.md (hunk 1 above)
Comment (ttomsu) [inline docs/commands.md]: Done.

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java (hunk 2 above)
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java]: Edited content a bit. Is that more clear?

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java (hunk 3 above)
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/EditOAuth2Command.java]: `UserInfoRequirements` extends `HashMap<String, String>`, which is great for two reasons:

1. It's required to use `DynamicParameter` 
1. Both key and value must be define by the user in the config.

**Pull request merged**